model/infinispan/src/main/java/org/keycloak/cluster/infinispan/InfinispanClusterProviderFactory.java

```diff
@@ -201,15 +203,25 @@ public class ViewChangeListener {
 
         @Merged
         public void mergeEvent(MergeEvent event) {
-           // During split-brain only Keycloak instances contained within the same partition will receive updates via
-           // the work cache. On split-brain heal it's necessary for us to clear all local caches so that potentially
-           // stale values are invalidated and subsequent requests are forced to read from the DB.
-           localExecutor.execute(() ->
-              Arrays.stream(InfinispanConnectionProvider.LOCAL_CACHE_NAMES)
-                    .map(name -> workCache.getCacheManager().getCache(name))
-                    .filter(cache -> cache.getCacheConfiguration().clustering().cacheMode() == CacheMode.LOCAL)
-                    .forEach(Cache::clear)
-           );
+            // During split-brain only Keycloak instances contained within the same partition will receive updates via
+            // the work cache. On split-brain healing, it's necessary for us to clear all local caches so that potentially
+            // stale values are invalidated and subsequent requests are forced to read from the DB.
+            localExecutor.execute(() ->
+                    Arrays.stream(InfinispanConnectionProvider.LOCAL_CACHE_NAMES)
+                            .map(name -> workCache.getCacheManager().getCache(name))
+                            .filter(cache -> cache.getCacheConfiguration().clustering().cacheMode() == CacheMode.LOCAL)
+                            .forEach(Cache::clear)
+            );
+
+            if (MultiSiteUtils.isPersistentSessionsEnabled()) {
```

Comment:
```suggestion
            if (Profile.isFeatureEnabled(Profile.Feature.PERSISTENT_USER_SESSIONS)) {
```
Those caches do not exist in multi-site deployments.